File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int number, int limit, int power) {
            int answer = -1;
        vector<int> num(number + 1, 1);
        for (int i = 1; i <= number; i++)
        {
            for (int j = 1; j <= i / 2; j++)
            {
                if (i%j == 0) num[i]++;
            }
        }

        for (auto n : num)
        {
            if (n > limit) answer += power;
            else answer += n;
        }
    return answer;
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cs
using System;

public class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        for(int i = 1; i<=number;i++)
        {
            if(i == 1)
            {
                answer +=1;
                continue;
            }
            int sum = 2;
            for(int j =2; j<=i/2;j++)
            {
                if(i%j == 0)
                    sum++;
            }
            if(sum > limit) sum = power;
            answer += sum;    
        }
        
        return answer;
    }
}